Patches/Rim Contractors Arsenal/ThingDefs_Misc/RangedContractors.xml

```diff
@@ -380,9 +411,435 @@
+					<aiUseBurstMode>TRUE</aiUseBurstMode>
+					<aiAimMode>AimedShot</aiAimMode>
+				</FireModes>
+			</li>
+			
+			<!--ML2375 Devastator, 12x64mech-->
+			
+			<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+				<defName>Gun_ML237S</defName>
+				<statBases>
+					<Bulk>12</Bulk>
+					<Mass>13.50</Mass>
+					<RangedWeapon_Cooldown>0.37</RangedWeapon_Cooldown>
+					<SightsEfficiency>3.51</SightsEfficiency>
+					<SwayFactor>2.67</SwayFactor>
+					<ShotSpread>0.06</ShotSpread>
+					<WorkToMake>59000</WorkToMake>
+				</statBases>
+				<costList>
+					<Chemfuel>20</Chemfuel>
+					<Steel>80</Steel>
+					<Plasteel>40</Plasteel>
+					<ComponentIndustrial>11</ComponentIndustrial>
+				</costList>
+				<Properties>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Bullet_12x64mmCharged</defaultProjectile>
+					<warmupTime>3.35</warmupTime>
+					<range>86</range>
+					<soundCast>Shot_HeavySniperContractors</soundCast>
+					<soundCastTail>GunTail_Medium</soundCastTail>
+					<muzzleFlashScale>12</muzzleFlashScale>
+				</Properties>
+				<AmmoUser>
+					<magazineSize>5</magazineSize>
+					<reloadTime>4</reloadTime>
+					<ammoSet>AmmoSet_12x64mmCharged</ammoSet>
+				</AmmoUser>
+				<FireModes>
+					<aiAimMode>AimedShot</aiAimMode>
+				</FireModes>
+			</li>
+			
+			<!--Phoenix - Charge 12 Gauge-->
+			
+			<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
+				<defName>Gun_HKCAS42A</defName>
+				<statBases>
+					<Bulk>7.62</Bulk>
+					<Mass>3.70</Mass>
+					<RangedWeapon_Cooldown>0.39</RangedWeapon_Cooldown>
+					<SightsEfficiency>1.10</SightsEfficiency>
+					<SwayFactor>1.13</SwayFactor>
+					<ShotSpread>0.08</ShotSpread>
+					<WorkToMake>48000</WorkToMake>
+				</statBases>
+				<costList>
+					<Chemfuel>10</Chemfuel>
+					<Steel>45</Steel>
+					<Plasteel>25</Plasteel>
+					<ComponentIndustrial>10</ComponentIndustrial>
+				</costList>
+				<Properties>
+					<recoilAmount>2.88</recoilAmount>
+					<verbClass>CombatExtended.Verb_ShootCE</verbClass>
+					<hasStandardCommand>true</hasStandardCommand>
+					<defaultProjectile>Bullet_12GaugeCharged</defaultProjectile>
+					<warmupTime>0.5</warmupTime>
+					<range>25</range>
+					<burstShotCount>3</burstShotCount>
+					<ticksBetweenBurstShots>14</ticksBetweenBurstShots>
+					<soundCast>Shot_LNF35L</soundCast>
+					<soundCastTail>GunTail_Medium</soundCastTail>
+					<muzzleFlashScale>11</muzzleFlashScale>
+				</Properties>
+				<AmmoUser>
+					<magazineSize>10</magazineSize>
+					<reloadTime>4.5</reloadTime>
+					<ammoSet>AmmoSet_12GaugeCharged</ammoSet>
+				</AmmoUser>
+				<FireModes>
+					<aimedBurstShotCount>2</aimedBurstShotCount>
+					<aiUseBurstMode>FALSE</aiUseBurstMode>
+					<aiAimMode>Snapshot</aiAimMode>
+				</FireModes>
+			</li>
+			
+			<!--SCR Survival Rifle-->
+			
+			<li Class="CombatExtended.PatchOperationMakeGunCECompatible">
```

Comment:
I can't say I'm a big fan of giving the weapon infinite ammo. Perhaps you could just make it accept a variety of different ammo types, instead? (which would be accomplished by creating a custom ammoset for it)

Reply:
Ultimately, this is once of those instances where I feel that if we had a recharging ammo pool type thing, It'd be ideal.

That being said, the issue with patching in a new ammo set is that it could:
A) Just use a single bullet type
In which instance it feels kind of silly. The idea of the weapon is to pretty much cut off the need for ammo supply. The description itself states that it needs no ammo to function. But that being said, if it did use one ammo type, it'd run into the issue where noone would have the appropriate ammo to run it it, since you'd be looking at the average thing having a myriad of possible enemies carrying different ammo. It'd make it no different than any other charge rifle, just needing one specific type of regular ammo rather than charge bullets, which undermines the general survival aspect of the weapon.
B)Use multiple bullet types
The issue here is that it either has to have a metric ton of side patches to check for something that uses that ammo (which might not even be possible via XML), or it'd force every possible ammo it could use to spawn soley for its own usage, even if they couldn't be used in any other gun, which would severly bloat the size of the ammo list even if the player isn't using a large variety of mods.

Ulitimately, I feel that giving one or two guns the ability to fire an infinite amount of ammo is fine. By building this instead of a conventional charge weapon, you're sacrificing a serious spread of stats in exchange for general ammo logevity, but if you can make charge weapons in the first place, aquiring ammo for them will almost certainly not be a serious issue.

I would consider tweaking the weapon tags/recipe/cost so that it is difficult to aquire without constructing it yourself, if you think that would resolve the issue, perhaps?